doc/design/GeneratedClientDesign.md

```diff
@@ -0,0 +1,133 @@
+
+### Enums
+
+Enums may be generated as a CLR enum or a C# "Strongly-typed string" as follows.
+
+- Enums are generated in the `.Models` namespace
+- A Cadl `enum` not found in a corresponding  `@knownValues` decorator is generated as a C# `enum`
+- Generated enum has corresponding internal extensions file providing conversions to/from string.
+- A Cadl string model with a `@knownValues` decorator is generated as a C# "Strongly-typed string" (sometimes called "extensible enum") and the corresponding Cadl `enum` is not generated as a C# `enum`
+
+GitHub issue: [autorest.csharp #2477](https://github.com/Azure/autorest.csharp/issues/2477)
+
+See examples:
+
+- CLR enum:
+  - [DayOfTheWeek.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-models-enum-properties/sdk/template/Azure.Template/src/Generated/Models/DayOfTheWeek.cs)
+  - [DayOfTheWeek.Serialization.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-models-enum-properties/sdk/template/Azure.Template/src/Generated/Models/DayOfTheWeek.Serialization.cs)
+- Stronly typed string (extensible enum):
+  - [TranslationLanguage.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-models-enum-properties/sdk/template/Azure.Template/src/Generated/Models/TranslationLanguage.cs)
+  - No corresponding .Serialization.cs file.
+
+### Nested Models
+
+Models that are properties of other models are called nested models.  A model that appears only as a non-root of a model graph is defined as partial class according to the same rules as root models.
+
+GitHub issue: [autorest.csharp #2489](https://github.com/Azure/autorest.csharp/issues/2489)
+
+See examples:
+
+- [RoundTripModel.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-models-nested-models/sdk/template/Azure.Template/src/Generated/Models/RoundTripModel.cs)
+- [RoundTripModel.Serialization.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-models-nested-models/sdk/template/Azure.Template/src/Generated/Models/RoundTripModel.Serialization.cs)
+
+Requirements for model graphs with circular dependencies will be defined in a later iteration.  GitHub issue [autorest.csharp #2506](https://github.com/Azure/autorest.csharp/issues/2506)
+
+#### Nested Model Shape
+
+Models that appear as properties of other models have their shape determined as follows:
+
+- Models that only appear as properties of input models are input models
+- Models that only appear as properties of output models are output models
+- Models that only appear as properties of round-trip models are round-trip models
+- Models that appear as properties of more than one model shape are round-trip models
+
+### Model Reference Documentation
+
+### Miscellaneous
+
+- Models should be generated with the `#nullable disable` because Azure SDK libraries have not yet exposed nullable reference types outside of Azure.Core.
```

Comment:
This is because we don't have tools to ensure that APIs would stay compatible in this regard.